### PR TITLE
Enhancement/deployment plan update

### PR DESCRIPTION
**Pull request by Yunnigg**

### CODE REVIEW

File: pronto_mia/lib/core/services/graphql_service.dart [hunk 1]
```diff
@@ -83,4 +98,19 @@ class GraphQLService {
       return '';
     }
   }
+
+  bool _isNetworkAvailable(OperationException exception) {
+    if (exception.linkException is NetworkException) {
+      return false;
+    }
+
+    if (exception.linkException is ServerException) {
```
Comment (Vec7or) [inline pronto_mia/lib/core/services/graphql_service.dart:107]: Why is this necessary? Is the exception thrown not always a Network exception?

File: pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart [hunk 2]
```diff
@@ -10,15 +14,22 @@ import 'package:pronto_mia/ui/views/deployment_plan/edit/deployment_plan_edit_vi
...
     with $DeploymentPlanEditView {
   final DeploymentPlan deploymentPlan;
 
-  DeploymentPlanEditView({Key key, this.deploymentPlan}) : super(key: key);
+  DeploymentPlanEditView({Key key, this.deploymentPlan}) : super(key: key) {
+    if (deploymentPlan != null) {
+      descriptionController.text = deploymentPlan.description;
+      availableFromController.text = deploymentPlan.availableFrom.toString();
+      availableUntilController.text = deploymentPlan.availableUntil.toString();
+      pdfPathController.text = 'upload.pdf';
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart:30]: If we want to implement the edit functionality this way the user should nonetheless have a possibility to view the currently uploaded file. Unfortunately in edit mode the user cannot click on the filename like he is used to from creation.

File: pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_viewmodel.dart [hunk 3]
```diff
@@ -1,31 +1,40 @@
...
 import 'package:pronto_mia/ui/views/deployment_plan/edit/deployment_plan_edit_view.form.dart';
+import 'package:pronto_mia/core/models/file_upload.dart';
+import 'package:pronto_mia/core/factories/error_message_factory.dart';
+import 'package:pronto_mia/core/models/deployment_plan.dart';
 
 class DeploymentPlanEditViewModel extends FormViewModel {
   DeploymentPlanService get _deploymentPlanService =>
       locator.get<DeploymentPlanService>();
   NavigationService get _navigationService => locator.get<NavigationService>();
+  ErrorMessageFactory get _errorMessageFactory =>
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_viewmodel.dart:17]: Why do we have to get this class over the locator in the first place? Isn't it a static helper class? So we could make getErrorMessage static?

File: pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_view.dart [hunk 4]
```diff
@@ -22,71 +22,93 @@ class DeploymentPlanOverviewView extends StatelessWidget {
...
-            ),
-            IconButton(
-              icon: const Icon(Icons.account_circle),
-              tooltip: 'Benutzerprofil',
-              onPressed: () {},
-            ),
-          ],
+          automaticallyImplyLeading: false,
+          title: const Text(
+            'Einsatzpläne',
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_view.dart:27]: Is the same title for both edit and normal view.

File: pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_viewmodel.dart [hunk 5]
```diff
@@ -10,25 +8,32 @@ import 'package:pronto_mia/app/app.router.dart';
...
 import 'package:pronto_mia/core/factories/error_message_factory.dart';
+import 'package:pronto_mia/core/services/logging_service.dart';
 
 class DeploymentPlanOverviewViewModel
     extends FutureViewModel<List<DeploymentPlan>> {
   DeploymentPlanService get _deploymentPlanService =>
       locator.get<DeploymentPlanService>();
-  NavigationService get _navigationService => locator<NavigationService>();
+  NavigationService get _navigationService => locator.get<NavigationService>();
   ErrorMessageFactory get _errorMessageFactory =>
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_viewmodel.dart:18]: Same as above

File: pronto_mia/lib/ui/views/login/login_view.dart [hunk 6]
```diff
@@ -18,60 +19,41 @@ class LoginView extends StatelessWidget with $LoginView {
...
-            children: [
-              Image.asset(
+        body: Column(
+          children: [
+            Container(
+              padding: const EdgeInsets.all(16.0),
+              child: Image.asset(
                 'assets/images/pronto_logo.png',
-                fit: BoxFit.contain,
+                height: 200,
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/login/login_view.dart:28]: Does unfortunately not help with the pixel problem.

File: pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart [hunk 7]
```diff
@@ -27,90 +38,100 @@ class DeploymentPlanEditView extends StatelessWidget
...
+            onTap: model.submitForm,
+            isBusy: model.busy(model.editBusyKey),
+          ),
+          secondaryButton: (() {
+            if (deploymentPlan != null) {
+              return ButtonSpecification(
+                title: 'Löschen',
+                onTap: model.removeDeploymentPlan,
+                isBusy: model.busy(model.removeBusyKey),
+                backgroundColor: Colors.red,
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart:129]: I would adjust the ButtonSpecification to have a boolean argument `isDestructive` which will then set the color accordingly instead of defining colors all over the place.

File: pronto_mia/lib/ui/views/startup/startup_viewmodel.dart [hunk 8]
```diff
@@ -1,28 +1,43 @@
...
+import 'package:pronto_mia/core/factories/error_message_factory.dart';
+import 'package:pronto_mia/core/services/logging_service.dart';
 
 class StartUpViewModel extends BaseViewModel {
   AuthenticationService get _authenticationService =>
       locator.get<AuthenticationService>();
   NavigationService get _navigationService => locator.get<NavigationService>();
   Future<LoggingService> get _loggingService =>
       locator.getAsync<LoggingService>();
+  ErrorMessageFactory get _errorMessageFactory =>
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/startup/startup_viewmodel.dart:18]: Same as above

**Review by Vec7or — CHANGES_REQUESTED** [review]

File: pronto_mia/lib/core/services/graphql_service.dart (hunk 1 above)
Comment (Yunnigg) [reply pronto_mia/lib/core/services/graphql_service.dart:107]: No, when I go offline on my end I get a ServerException with an empty parsedResponse property.

File: pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_view.dart (hunk 4 above)
Comment (Yunnigg) [reply pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_view.dart:27]: Fixed.

File: pronto_mia/lib/ui/views/login/login_view.dart (hunk 6 above)
Comment (Yunnigg) [reply pronto_mia/lib/ui/views/login/login_view.dart:28]: I suggest that you have a look at this, since I can't reproduce the problem on my machine anymore.

File: pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart (hunk 7 above)
Comment (Yunnigg) [reply pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart:129]: Added isDestructive in favor of backgroundColor.

File: pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart (hunk 2 above)
Comment (Yunnigg) [reply pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart:30]: This was not intended. A implented a fix.

File: pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_viewmodel.dart (hunk 3 above)
Comment (Yunnigg) [reply pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_viewmodel.dart:17]: As discussed we can implement the method static, but this will be done in a later pull request.

**Review by Vec7or — APPROVED** [review]
